modin/pandas/test/utils.py

```diff
@@ -478,8 +478,14 @@ def df_equals(df1, df2):
     # Convert to pandas
     if isinstance(df1, (pd.DataFrame, pd.Series)):
         df1 = to_pandas(df1)
+        # Get rid of implementation detail:
```

Comment:
Not sure if this is perfectly fine, a necessary hack, or a really bad idea.

Reply:
why do we need these changes? Isn't `Series` automatically do this (if it handled correctly)?
https://github.com/modin-project/modin/blob/e640ecd8cb785375720b8a817eb48a2b5f3a88ad/modin/pandas/series.py#L1700-L1705

Reply:
It wasn't working, the `__reduced__` was leaking. But that at least gives me a hint of where to start looking to fix this better.